src/sentry/db/models/base.py

```diff
@@ -155,6 +155,74 @@ class Meta:
 
     __repr__ = sane_repr("id")
 
+    def save(self, *args: Any, **kwds: Any) -> None:
+        from sentry.models.outbox import outbox_context
+
+        from .outboxes import ProducesControlOutboxesOnUpdate, ProducesRegionOutboxOnUpdate
+
+        if isinstance(self, ProducesControlOutboxesOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                super().save(*args, **kwds)
+                for outbox in self.outboxes_for_update():
+                    outbox.save()
+        elif isinstance(self, ProducesRegionOutboxOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                super().save(*args, **kwds)
+                self.outbox_for_update().save()
+        else:
+            super().save(*args, **kwds)
+
+    def update(self, *args: Any, **kwds: Any) -> None:
+        from sentry.models.outbox import outbox_context
+
+        from .outboxes import ProducesControlOutboxesOnUpdate, ProducesRegionOutboxOnUpdate
+
+        if isinstance(self, ProducesControlOutboxesOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                super().update(*args, **kwds)
+                for outbox in self.outboxes_for_update():
+                    outbox.save()
+        elif isinstance(self, ProducesRegionOutboxOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                super().update(*args, **kwds)
+                self.outbox_for_update().save()
+        else:
+            super().update(*args, **kwds)
+
+    def delete(self, *args: Any, **kwds: Any) -> None:
+        from sentry.models.outbox import outbox_context
+
+        from .outboxes import ProducesControlOutboxesOnUpdate, ProducesRegionOutboxOnUpdate
+
+        if isinstance(self, ProducesControlOutboxesOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                for outbox in self.outboxes_for_update():
+                    outbox.save()
+        elif isinstance(self, ProducesRegionOutboxOnUpdate):
+            with outbox_context(
+                transaction.atomic(router.db_for_write(type(cast(Any, self)))),
+                flush=self.__default_flush__,
+            ):
+                self.outbox_for_update().save()
+                super().delete(*args, **kwds)
+        else:
+            super().delete(*args, **kwds)
```

Comment:
`update` and `delete` have return values that may be worth preserving.

Reply:
Roger, good call out.